src/popup/popup.js

```
@@ -12,12 +12,99 @@
+            url.includes(pattern.pattern);
+    } else {
+        const pattern = URL_MATCHERS[patternKey];
+        return typeof pattern === 'string' ?
+            url.includes(pattern) : false;
+    }
+}
+
+// Add this function to extract query parameters from URL
+function getUrlParameter(url, name) {
+    name = name.replace(/[\[]/, '\\[').replace(/[\]]/, '\\]');
+    const regex = new RegExp('[\\?&]' + name + '=([^&#]*)');
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)